src/layouts/app/connection-mixin.js

```diff
@@ -155,6 +157,13 @@ export default (superClass) =>
         getState()
       );
 
+      getUserLanguage(this.hass).then((language) => {
+        this.hass.selectedLanguage = language;
+        storeState(this.hass);
```

Comment:
Only update storage if the selected language was different from the currently selected language.